### PR TITLE
feat(add): readme for cassandra installation using kudo

### PR DESCRIPTION
**Pull request by prabhatkumarthakur**
- In this PR I have added a readme that describes the use case, Cassandra deployment using Kudo with openebs device storage class.
- Modified the folder structure

## Folder structure
```
continuous-kubernetes
├── eks
│   ├── cassandra-deploy
│   │   ├── assert-cassandra-pod.yaml
│   │   ├── cassandra-inference.yaml
│   │   ├── cassandra_install.sh
│   │   ├── cassandra_uninstall.sh
│   │   └── README.md
│   ├── eks-setup
│   │   ├── cluster.yaml
│   │   ├── eks_install.sh
│   │   └── eks_uninstall.sh
│   ├── experiments
│   │   ├── assert-kubesystem-pod.yaml
│   │   ├── assert-openebs-pod.yaml
│   │   ├── ci.yml
│   │   └── inference.yaml
│   └── openebs
│       ├── disk.yaml
│       ├── openebs_install.sh
│       ├── openebs_uninstall.sh
│       ├── storageclass.yaml
│       ├── volumeattachment.sh
│       └── volumeattachment_template.yaml
├── kudo-operator
│   ├── assert-cert-manger-pod.yaml
│   ├── cert-manager-inference.yaml
│   ├── kudo_install.sh
│   └── kudo_uninstall.sh
├── LICENSE
├── Makefile
└── README.md
```

### CODE REVIEW

File: .github/workflows/action.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Github action to create EKS cluster using eksctl 
+name: Github action to create EKS cluster and deploy cassandra sts
```
Comment (ranjithwingrider) [inline .github/workflows/action.yml:1]: cassandra sts ->  Cassandra StatefulSet

File: eks/cassandra-deploy/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+## Cassandra deployment using kudo operator
+
+In this usecase below activity have been performed:
```
Comment (ranjithwingrider) [inline eks/cassandra-deploy/README.md:3]: In this usecase below activity have been performed:- > 
In this use case , the following steps are performed

File: eks/cassandra-deploy/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+- Create a 3 node EKS cluster in ap-south-1 region
```
Comment (ranjithwingrider) [inline eks/cassandra-deploy/README.md:5]: ap-south-1 -> `ap-south-1`

File: eks/cassandra-deploy/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+- Install OpenEBS latest version
+- Create an Amazon csi volume and use the same as a block devices attached to each node
```
Comment (ranjithwingrider) [inline eks/cassandra-deploy/README.md:7]: Create an Amazon csi volume and use the same as a block devices attached to each node -> 
Create an Amazon EBS volume and attach it to each node.

File: eks/cassandra-deploy/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- Deploy cert-manager
+- Install Kudo-operator
+- Deploy Cassandra statefulset using kudo operator and openebs-device as storage class
```
Comment (ranjithwingrider) [inline eks/cassandra-deploy/README.md:10]: openebs-device -> `openebs-device`

Comment (ranjithwingrider) [inline eks/cassandra-deploy/README.md:10]: storage class -> `Storage Class`

File: eks/cassandra-deploy/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+- Check the status of cassandra sts pods.
```
Comment (ranjithwingrider) [inline eks/cassandra-deploy/README.md:11]: cassandra sts -> Cassandra

File: eks/cassandra-deploy/README.md (hunk 4 above)
Comment (gprasath) [reply eks/cassandra-deploy/README.md:7]: Is the number of volumes based on number of nodes?

File: eks/cassandra-deploy/README.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+## Cassandra deployment using kudo operator
+
+In this usecase below activity have been performed:
+
+- Create a 3 node EKS cluster in ap-south-1 region
+- Install OpenEBS latest version
+- Create an Amazon csi volume and use the same as a block devices attached to each node
+- Deploy cert-manager
```
Comment (gprasath) [inline eks/cassandra-deploy/README.md:8]: Is cert-manager , a prerequisite for kudo-operator deployment?

File: eks/cassandra-deploy/README.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+- Install Kudo-operator
+- Deploy Cassandra statefulset using kudo operator and openebs-device as storage class
+- Check the status of cassandra sts pods.
+
+### Cleanup phase:
+- Uninstall Cassandra statefulset
+- Uninstall Kudo operator
+- Delete the disks
```
Comment (gprasath) [inline eks/cassandra-deploy/README.md:16]: aren't we deleting pvc?

File: eks/cassandra-deploy/cassandra_install.sh [hunk 9]
```diff
@@ -10,11 +10,11 @@ kubectl create ns cassandra
 
 echo -e "\nInstall cassandra operator with kudo"
 kubectl kudo install cassandra --namespace=$namespace_name --instance $instance_name -p NODE_STORAGE_CLASS=openebs-device
-sleep 45
+sleep 35
```
Comment (gprasath) [inline eks/cassandra-deploy/cassandra_install.sh:13]: how many replicas this sts has? It should be configurable

Comment (gprasath) [reply eks/cassandra-deploy/cassandra_install.sh:13]: why do we sleep? Instead we need to check if the kudo-operator has started running

**Review by gprasath — CHANGES_REQUESTED** [review]

File: .github/workflows/action.yml (hunk 1 above)
Comment (prabhatkumarthakur) [reply .github/workflows/action.yml:1]: done

File: eks/cassandra-deploy/README.md (hunk 2 above)
Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/README.md:3]: done

File: eks/cassandra-deploy/README.md (hunk 3 above)
Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/README.md:5]: @gprasath yes,we can make changes in the cluster.yaml to deploy the cluster in other regions.

File: eks/cassandra-deploy/README.md (hunk 4 above)
Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/README.md:7]: yes, the script had been written to create 3 volumes for 3 nodes.
@gprasath  I have followed the below doc and automated the same. The aim was to run everything from a laptop or Github action without any manual intervention.
https://mayadata.io/workloads/cassandra-on-kubernetes

File: eks/cassandra-deploy/README.md (hunk 5 above)
Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/README.md:10]: done

Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/README.md:10]: done

File: eks/cassandra-deploy/README.md (hunk 6 above)
Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/README.md:11]: done

File: Makefile [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
 .PHONY: all
 eks-install: 
-	@cd eks && ./eks_install.sh
+	@cd eks/eks-setup && ./eks_install.sh
 openebs-install: 
-	@cd openebs && ./openebs_install.sh
+	@cd eks/openebs && ./openebs_install.sh
 kudo-install: 
-	@cd kudo-operator && ./kudo_install.sh
+	@cd eks/kudo-operator && ./kudo_install.sh
```
[the remark this answers is not in the document]
Comment (prabhatkumarthakur) [reply Makefile:8]: I will move kudo-operator outside eks folder

File: eks/cassandra-deploy/README.md (hunk 7 above)
Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/README.md:8]: yes, As per the workload article it is required.

File: eks/cassandra-deploy/README.md (hunk 8 above)
Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/README.md:16]: yes, we are deleting it in openebs uninstallation step.

File: eks/cassandra-deploy/cassandra_install.sh (hunk 9 above)
Comment (prabhatkumarthakur) [reply eks/cassandra-deploy/cassandra_install.sh:13]: The default replica count for Cassandra is 3.
We are checking the kudo-operator running status in kundo_install.sh script.

File: eks/cassandra-deploy/README.md (hunk 5 above)
Comment (ranjithwingrider) [reply eks/cassandra-deploy/README.md]: Try to use the same word everywhere for Storage Class instead of storage class.
Also change kudo to Kudo

Comment (ranjithwingrider) [reply eks/cassandra-deploy/README.md]: Use Kudo instead of kudo everywhere.

File: eks/cassandra-deploy/README.md (hunk 6 above)
Comment (ranjithwingrider) [reply eks/cassandra-deploy/README.md]: cassandra  -> Cassandra

**Review by ranjithwingrider — APPROVED** [review]
Looks good with the content.

**Review by gprasath — APPROVED** [review]

**Pull request merged**